lib/remap-functions/src/to_string.rs

```diff
@@ -58,21 +50,12 @@ impl Expression for ToStringFn {
             Regex(v) => Ok(v.to_string().into()),
             Null => Ok("".into()),
             Map(_) | Array(_) => Err("unable to convert value to string".into()),
-        };
-
-        crate::util::convert_value_or_default(
-            self.value.execute(state, object),
-            self.default.as_ref().map(|v| v.execute(state, object)),
-            to_string,
-        )
+        }
     }
 
     fn type_def(&self, state: &state::Compiler) -> TypeDef {
         self.value
             .type_def(state)
-            .merge_with_default_optional(
-                self.default.as_ref().map(|default| default.type_def(state)),
-            )
             .with_constraint(value::Kind::Bytes)
```

Comment:
This has to have an `fallible_unless` line, since maps and arrays always make this function fallible.